litmus-portal/graphql-server/pkg/chaos-workflow/handler/handler.go

```diff
@@ -253,6 +258,37 @@ func QueryListWorkflowByIDs(workflow_ids []*string) ([]*model.Workflow, error) {
 	return result, nil
 }
 
+//ReRunWorkflow sends workflow run request(single run workflow only) to agent on workflow re-run request
+func ReRunWorkflow(workflowID string) (string, error) {
+	query := bson.D{{"workflow_id", workflowID}, {"isRemoved", false}}
+	workflows, err := database.GetWorkflows(query)
+	if err != nil {
+		log.Print("Could not get workflow :", err)
+		return "could not get workflow", err
+	}
+	if len(workflows) == 0 {
+		return "", errors.New("no such workflow found")
+	}
+	resKind := gjson.Get(workflows[0].WorkflowManifest, "kind").String()
```

Comment:
Right now, we have the Argo workflow struct. Can we parse it with that argo struct?

Reply:
argo struct pkg is already there in go.mod.